modules/phase_field/src/postprocessors/GrainTracker.C

```diff
@@ -143,20 +147,84 @@ GrainTracker::finalize()
 
   expandHalos();
 
-  FeatureFloodCount::communicateAndMerge();
+  // Build up the grain map on the root processor
+  communicateAndMerge();
+
+  /**
+   * Track Grains
+   */
+  Moose::perf_log.push("trackGrains()", "GrainTracker");
+  std::vector<unsigned int> new_grain_indices;
+
+  // Track grains (only on the root processor)
+  if (_is_master)
+    trackGrains(new_grain_indices);
+
+  if (!new_grain_indices.empty())
+  {
+    // Communicate new grain indices with the remaining processors
+    auto new_grains_count = new_grain_indices.size();
+    _communicator.broadcast(new_grains_count);
+
+    new_grain_indices.resize(new_grains_count);
+    _communicator.broadcast(new_grain_indices);
+
+    // Trigger the callback on all processors
```

Comment:
This is a decent tutorial:  http://mpitutorial.com/tutorials/dynamic-receiving-with-mpi-probe-and-mpi-status/

Scroll down to `Using MPI_Probe to find out the message size`

This is also how I do it in the new nonblocking packed range stuff I added (although, of course, I'm doing the non-blocking version of all of this).

The basic idea is that you do:

```
MPI_Probe(status, message_type)
MPI_Get_count(status, count)
resize_receive_buffer(buffer, count)
MPI_Recv(buffer, count)
```